pkg/datapath/linux/config/config.go

```diff
@@ -356,10 +356,7 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 			if option.Config.EnablePMTUDiscovery {
 				cDefinesMap["ENABLE_DSR_ICMP_ERRORS"] = "1"
 			}
-			if cfg.LBConfig.LBMode == loadbalancer.LBModeHybrid {
-				cDefinesMap["ENABLE_DSR_HYBRID"] = "1"
-			} else if cfg.LBConfig.LBModeAnnotation {
-				cDefinesMap["ENABLE_DSR_HYBRID"] = "1"
+			if cfg.LBConfig.LBMode == loadbalancer.LBModeHybrid || cfg.LBConfig.LBModeAnnotation {
```

Comment:
I think this should be part of the second patch. Then we can easily backport the first patch to `v1.18`, and ensure that all service entries are set up for a smooth upgrade to `v1.19`.